src/cogent3/core/sequence.py

```diff
@@ -1906,35 +1906,35 @@ def _input_vals_neg_step(seqlen, start, stop, step):
     return (0, 0, 1) if start < stop else (start, stop, step)
 
 
-class SeqView:
-    __slots__ = ("seq", "start", "stop", "step", "_offset", "_seqid", "_seq_len")
+class SliceRecordABC(ABC):
+    """Abstract base class for recording the history of operations to be applied
+    to some underlying data
 
-    def __init__(
-        self,
-        seq,
-        *,
-        start: Optional[int] = None,
-        stop: Optional[int] = None,
-        step: Optional[int] = None,
-        offset: int = 0,
-        seqid: Optional[str] = None,
-        seq_len: Optional[int] = None,
-    ):
-        if step == 0:
-            raise ValueError("step cannot be 0")
-        step = 1 if step is None else step
+    Notes
+    -----
+    _seq_len refers to the Python definiton of a sequence (i.e., array, str, etc).
 
-        func = _input_vals_pos_step if step > 0 else _input_vals_neg_step
-        start, stop, step = func(len(seq), start, stop, step)
-        self.seq = seq
-        self.start = start
-        self.stop = stop
-        self.step = step
-        self._offset = offset
-        self._seqid = seqid
-        if seq_len is not None and seq_len != len(seq):
-            raise AssertionError(f"{seq_len} != {len(self.seq)})")
-        self._seq_len = seq_len or len(self.seq)
+    Expected constructor signature:
+    ``__init__(self, seq: Any, *, start: int = None, stop: int = None, step: int = None, **kwargs: Any)``
+    """
+
+    seq: Any
+    start: int
+    stop: int
+    step: int
+    _seq_len: int
+    _offset: int
+
+    @abstractmethod
+    def copy(self): ...
+
+    @property
+    def _zero_slice(self):
+        return self.__class__(seq=type(self.seq)())
```

Comment:
Make this an abstract property and leave it empty in this abstract base class. For example...
```python
@property
@abstractmethod
def _zero_slice(self): ...
```